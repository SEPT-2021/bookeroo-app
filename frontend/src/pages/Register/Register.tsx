import React, { useState } from "react";
import axios from "axios";
import {
  Avatar,
  Box,
  Button,
  Checkbox,
  createStyles,
  FormControlLabel,
  Grid,
  Hidden,
  Link,
  Paper,
  TextField,
  Theme,
  Typography,
  withStyles,
  WithStyles,
} from "@material-ui/core";
import { LockOutlined } from "@material-ui/icons";
import logo from "../../assets/logo.svg";

function Register({ classes }: RegisterProps) {
  const [Username, setUsername] = useState("");
  const [FirstName, setFirstName] = useState("");
  const [LastName, setLastName] = useState("");
  const [Password, setPassword] = useState("");
  const [ConfirmPassword, setConfirmPassword] = useState("");

  async function registerPost() {
    axios({
      method: "POST",
      url: "/api/users/register",
      data: {
        username: Username,
        firstName: FirstName,
        lastName: LastName,
        password: Password,
        confirmPassword: ConfirmPassword,
      },
    }).then(
      (response) => {
        // eslint-disable-next-line no-console
        return(response.data.json)
      },
      (error) => {
        console.log(error);
      }
    );
  }

  return (
    <Grid container component="main" className={classes.root}>
      <Grid item xs={12} sm={4} md={7} className={classes.steps}>
        <Box
          display="flex"
          alignItems="center"
          justifyContent="center"
          flexWrap="wrap"
          width="100%"
        >
          <img src={logo} alt="logo" />
          <Hidden xsDown>
            <div>
              <Typography variant="h1">Bookeroo</Typography>
              <Typography variant="h5">
                Dealing books has never been this easy
              </Typography>
            </div>
          </Hidden>
        </Box>
      </Grid>
      <Grid item xs={12} sm={8} md={5} component={Paper} elevation={6} square>
        <div className={classes.paper}>
          <Avatar className={classes.avatar}>
            <LockOutlined />
          </Avatar>
          <Typography component="h1" variant="h5">
            Register
          </Typography>
          <Box>
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              id="email"
              label="Email Address"
              name="email"
              autoComplete="email"
              autoFocus
              onChange={(e) => setUsername(e.target.value)}
            />
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              name="firstName"
              label="First Name"
              id="firstName"
              autoComplete="firstName"
              onChange={(e) => setFirstName(e.target.value)}
            />
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              name="lastName"
              label="Last Name"
              id="lastName"
              autoComplete="lastName"
              onChange={(e) => setLastName(e.target.value)}
            />
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              name="password"
              label="Password"
              type="password"
              id="password"
              autoComplete="current-password"
              onChange={(e) => setPassword(e.target.value)}
            />
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              name="confirmPassword"
              label="Confirm Password"
              type="password"
              id="confirmPassword"
              autoComplete="current-password"
              onChange={(e) => setConfirmPassword(e.target.value)}
            />
            <FormControlLabel
              control={<Checkbox value="remember" color="primary" />}
              label="Remember me"
            />
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
              onClick={registerPost}
            >
              Register Now
            </Button>
            <Grid container>
              <Grid item>
                <Link href="/api/users/login" variant="body2">
                  Have an account? Login Now
                </Link>
              </Grid>
            </Grid>
            <Box mt={5}>
              <Typography variant="caption">
                Disclaimer: This application was built for SEPT 2021, and is a
                university project. Please do not provide valuable data into
                this application.
              </Typography>
            </Box>
          </Box>
        </div>
      </Grid>
    </Grid>
  );
}

const styles = (theme: Theme) =>
  createStyles({
    root: {
      height: "100vh",
    },
    steps: {
      "& img": {
        width: 230,
        margin: theme.spacing(6),
      },
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      flexDirection: "column",
    },
    paper: {
      margin: theme.spacing(8, 4),
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
    },
    avatar: {
      margin: theme.spacing(1),
      backgroundColor: theme.palette.secondary.main,
    },
    submit: {
      margin: theme.spacing(3, 0, 2),
    },
  });

type RegisterProps = WithStyles<typeof styles>;

export default withStyles(styles)(Register);
