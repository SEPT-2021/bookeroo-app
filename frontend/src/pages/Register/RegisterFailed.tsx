import React from "react";

import {
  createStyles,
  Grid,
  Theme,
  withStyles,
  WithStyles,
} from "@material-ui/core";

function RegisterFailed({ classes }: RegisterSuccessProps) {
  return (
    <Grid
      container
      spacing={0}
      direction="column"
      alignItems="center"
      justify="center"
      style={{ minHeight: "100vh" }}
    >
      <Grid className={classes.root}>
        <h1> FAILED </h1>
      </Grid>
    </Grid>
  );
}

const styles = (theme: Theme) =>
  createStyles({
    root: {
      height: "100vh",
    },
    steps: {
      "& img": {
        width: 230,
        margin: theme.spacing(6),
      },
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      flexDirection: "column",
    },
    paper: {
      margin: theme.spacing(8, 4),
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
    },
    avatar: {
      margin: theme.spacing(1),
      backgroundColor: theme.palette.secondary.main,
    },
    submit: {
      margin: theme.spacing(3, 0, 2),
    },
  });

type RegisterSuccessProps = WithStyles<typeof styles>;

export default withStyles(styles)(RegisterFailed);
