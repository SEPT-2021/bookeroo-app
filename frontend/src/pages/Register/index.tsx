import React, { useContext, useState } from "react";
import { Redirect } from "react-router-dom";
import { LockOutlined } from "@material-ui/icons";
import { useMutation } from "react-query";
import {
  Avatar,
  Box,
  createStyles,
  Grid,
  Hidden,
  Paper,
  Theme,
  Typography,
  withStyles,
  WithStyles,
} from "@material-ui/core";
import logo from "../../assets/logo.svg";
import Link from "../../util/Link";
import { registerUser } from "../../util/api";
import LoadingButton from "../../util/LoadingButton";
import FormField from "../../util/FormField";
import { GlobalContext } from "../../components/GlobalContext";

function Register({ classes }: RegisterProps) {
  const [username, setUsername] = useState("");
  const [firstName, setFirstName] = useState("");
  const [lastName, setLastName] = useState("");
  const [password, setPassword] = useState("");
  const [confirmPassword, setConfirmPassword] = useState("");
  const [addressLine1, setAddressLine1] = useState("");
  const [addressLine2, setAddressLine2] = useState("");
  const [city, setCity] = useState("");
  const [state, setState] = useState("");
  const [postalCode, setPostalCode] = useState("");
  const [phoneNumber, setPhoneNumber] = useState("");

  const context = useContext(GlobalContext);
  const passwordsMatchingError =
    password && password !== confirmPassword && "Passwords must match";
  const { data, error, isSuccess, isLoading, mutate } =
    useMutation(registerUser);

  const onSubmit = async () => {
    mutate({
      username,
      password,
      firstName,
      lastName,
      roles: "ROLE_USER",
      addressLine1,
      addressLine2,
      city,
      state,
      postalCode,
      phoneNumber,
      enabled: true,
    });
  };

  if (isSuccess && data) {
    context.login(data);
    return <Redirect to="/allBooks" />;
  }

  return (
    <Grid container component="main" className={classes.root}>
      <Grid item xs={12} sm={4} md={7} className={classes.steps}>
        <Box
          display="flex"
          alignItems="center"
          justifyContent="center"
          flexWrap="wrap"
          width="100%"
        >
          <img src={logo} alt="logo" />
          <Hidden xsDown>
            <div>
              <Typography variant="h1">Bookeroo</Typography>
              <Typography variant="h5">
                Dealing books has never been this easy
              </Typography>
            </div>
          </Hidden>
        </Box>
      </Grid>
      <Grid item xs={12} sm={8} md={5} component={Paper} elevation={6} square>
        <div className={classes.paper}>
          <Avatar className={classes.avatar}>
            <LockOutlined />
          </Avatar>
          <Typography component="h1" variant="h5">
            Register
          </Typography>
          <Box>
            <FormField
              errors={error?.response?.data}
              label="Email Address"
              name="username"
              autoComplete="email"
              onChange={setUsername}
            />
            <FormField
              errors={error?.response?.data}
              name="firstName"
              label="First Name"
              autoComplete="firstName"
              onChange={setFirstName}
            />
            <FormField
              errors={error?.response?.data}
              name="lastName"
              label="Last Name"
              autoComplete="lastName"
              onChange={setLastName}
            />
            <FormField
              errors={error?.response?.data}
              name="phoneNumber"
              label="Phone Number"
              autoComplete="phoneNumber"
              onChange={setPhoneNumber}
            />
            <FormField
              errors={error?.response?.data}
              name="password"
              label="Password"
              type="password"
              autoComplete="current-password"
              onChange={setPassword}
            />
            <FormField
              errors={
                passwordsMatchingError
                  ? {
                      confirmPassword: passwordsMatchingError,
                    }
                  : undefined
              }
              name="confirmPassword"
              label="Confirm Password"
              type="password"
              autoComplete="current-password"
              onChange={setConfirmPassword}
            />
            <FormField
              errors={error?.response?.data}
              name="addressLine1"
              label="Address Line 1"
              autoComplete="addressLine1"
              onChange={setAddressLine1}
            />
            <FormField
              errors={error?.response?.data}
              name="addressLine2"
              label="Address Line 2"
              autoComplete="addressLine2"
              onChange={setAddressLine2}
            />
            <FormField
              errors={error?.response?.data}
              name="city"
              label="City"
              autoComplete="city"
              onChange={setCity}
            />
            <FormField
              errors={error?.response?.data}
              name="state"
              label="State"
              autoComplete="state"
              onChange={setState}
            />
            <FormField
              errors={error?.response?.data}
              name="postalCode"
              label="Postal Code"
              autoComplete="postalCode"
              onChange={setPostalCode}
            />

            <LoadingButton
              loading={isLoading}
              disabled={!!passwordsMatchingError}
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
              onClick={onSubmit}
            >
              Register
            </LoadingButton>
            <Grid container>
              <Grid item>
                <Link to="/login">Have an account? Login Now</Link>
              </Grid>
            </Grid>
            <Box mt={5}>
              <Typography variant="caption">
                Disclaimer: This application was built for SEPT 2021, and is a
                university project. Please do not provide valuable data into
                this application.
              </Typography>
            </Box>
          </Box>
        </div>
      </Grid>
    </Grid>
  );
}

const styles = (theme: Theme) =>
  createStyles({
    root: {
      height: "100vh",
    },
    steps: {
      "& img": {
        width: 230,
        margin: theme.spacing(6),
      },
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      flexDirection: "column",
    },
    paper: {
      margin: theme.spacing(15, 4),
      [theme.breakpoints.down("sm")]: {
        margin: theme.spacing(8, 4),
      },
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
    },
    avatar: {
      margin: theme.spacing(1),
      backgroundColor: theme.palette.secondary.main,
    },
    submit: {
      margin: theme.spacing(3, 0, 2),
    },
  });

type RegisterProps = WithStyles<typeof styles>;

export default withStyles(styles)(Register);
