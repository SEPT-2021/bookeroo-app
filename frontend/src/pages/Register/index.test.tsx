import React from "react";
import { render } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "react-query";
import Register from ".";

test("renders register page", () => {
  const { container } = render(
    <QueryClientProvider client={new QueryClient()}>
      <Register />
    </QueryClientProvider>
  );
  expect(container).toMatchSnapshot();
});
