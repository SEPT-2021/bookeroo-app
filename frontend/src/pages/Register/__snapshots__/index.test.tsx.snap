// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders register page 1`] = `
<div>
  <main
    class="MuiGrid-root Register-root-1 MuiGrid-container"
  >
    <div
      class="MuiGrid-root Register-steps-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-7"
    >
      <div
        class="MuiBox-root MuiBox-root-6"
      >
        <img
          alt="logo"
          src="logo.svg"
        />
      </div>
    </div>
    <div
      class="MuiPaper-root MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 MuiPaper-elevation6"
    >
      <div
        class="Register-paper-3"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular Register-avatar-4 MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <h1
          class="MuiTypography-root MuiTypography-h5"
        >
          Register
        </h1>
        <div
          class="MuiBox-root MuiBox-root-7"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
              data-shrink="true"
              for="email"
              id="email-label"
            >
              Email Address
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="email"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="email"
                name="email"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-10 PrivateNotchedOutline-legendNotched-11"
                >
                  <span>
                    Email Address
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
              for="firstName"
              id="firstName-label"
            >
              First Name
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="firstName"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="firstName"
                name="firstName"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-10"
                >
                  <span>
                    First Name
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
              for="lastName"
              id="lastName-label"
            >
              Last Name
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="lastName"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="lastName"
                name="lastName"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-10"
                >
                  <span>
                    Last Name
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
              for="password"
              id="password-label"
            >
              Password
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="current-password"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="password"
                name="password"
                required=""
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-10"
                >
                  <span>
                    Password
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
              for="confirmPassword"
              id="confirmPassword-label"
            >
              Confirm Password
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="current-password"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="confirmPassword"
                name="confirmPassword"
                required=""
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-10"
                >
                  <span>
                    Confirm Password
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-12 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-15"
                  data-indeterminate="false"
                  type="checkbox"
                  value="remember"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Remember me
            </span>
          </label>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained Register-submit-5 MuiButton-containedPrimary MuiButton-fullWidth"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Register Now
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
                href="/api/users/login"
              >
                Have an account? Login Now
              </a>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-16"
          >
            <span
              class="MuiTypography-root MuiTypography-caption"
            >
              Disclaimer: This application was built for SEPT 2021, and is a university project. Please do not provide valuable data into this application.
            </span>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
